client/my-sites/people/people-profile/index.jsx

```diff
@@ -101,17 +106,24 @@ class PeopleProfile extends React.PureComponent {
 		page( `/read/blogs/${ blogId }` );
 	};
 
-	renderName = () => {
-		const user = this.props.user;
+	renderNameOrEmail = () => {
+		const { translate, type, user } = this.props;
+
 		let name;
 		if ( ! user ) {
-			name = this.props.translate( 'Loading Users', {
+			name = translate( 'Loading Users', {
 				context: 'Placeholder text while fetching users.',
 			} );
 		} else if ( user.name ) {
 			name = user.name;
 		} else if ( user.label ) {
 			name = user.label;
+		} else if ( 'invite' === type && user.email ) {
+			// If an invite was sent to a WP.com user, the invite object will have
+			// either a display name (if set) or the WP.com username. Invites can
+			// also be sent to any email address, in which case the other details
+			// will not be set and we therefore display the user's email.
+			name = user.email;
```

Comment:
I think this should be `user.login || user.email` (and we should remove `user.email` from the `if` condition).  Otherwise, we don't handle the case where a WP.com user has a blank display name.